pkg/trace/api/api.go

```diff
@@ -464,6 +465,7 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		return
 	}
 
+	defer timing.Since("datadog.trace_agent.receiver.serve_traces_ms", time.Now())
```

Comment:
This is emitted on errors as well as on successful response writes. Is that intentional? Would it make sense to filter out the unsuccessful responses or tag the metric?

Reply:
You are right again. That was not intentional. The `timing.Since` API does not allow for tags, but I think we can refactor this to make it work like you are proposing. 

In the current state, it is possible for error responses to skew the results.

Reply:
Please check latest commit. Let me know if you think it's better.